qemu/tests/migration_with_json_backing.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import json
+
+from avocado.utils import process
+
+from virttest.qemu_storage import QemuImg, get_image_json, get_image_repr
+from virttest.data_dir import get_data_dir
+from virttest.qemu_devices.qdevices import QBlockdevFormatNode
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+def run(test, params, env):
+    """
+    KVM migration test:
+    1) Create a snapshot chain base->sn1->sn2 with backing:json format.
+    2) Start src guest with sn2.
+    3) Start dst guest with sn2 and listening status.
+    4) Do migration from src to dst.
+    5) After migration finished, check block info in dst.
+    6) In dst, do snapshot from sn2 to sn3
+    7) Check sn3 backing chain by qemu-img info
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def get_img_objs(images):
+        return [QemuImg(params.object_params(tag), get_data_dir(), tag)
+                for tag in images]
+
+    def _create_image_with_backing(image):
+        secids = [s.image_id for s in image.encryption_config.base_key_secrets]
+        if image.base_tag in secids:
+            image.create(params)
+        else:
+            # Hard code qemu-img cmd here
+            base_params = params.object_params(image.base_tag)
+            backing_file = "'%s'" % get_image_json(
+                    image.base_tag, base_params, get_data_dir())
+            cmd = '{cmd} create -F {base_fmt} -b {backing} -f {fmt} {f} {size}'
+            qemu_img_cmd = cmd.format(cmd=image.image_cmd,
+                                      base_fmt=base_params['image_format'],
+                                      backing=backing_file,
+                                      fmt=image.image_format,
+                                      f=image.image_filename,
+                                      size=image.size)
+
+            msg = "Create image by command: %s" % qemu_img_cmd
+            LOG_JOB.info(msg)
+            cmd_result = process.run(
+                qemu_img_cmd, shell=True, verbose=False, ignore_status=True)
```

Comment:
we didn't handle cmd_result, so I suggest we set ignore_status=False, ie. let VT throw an exception if anything wrong happens. Note that line35 will also throw an exception when hitting an error